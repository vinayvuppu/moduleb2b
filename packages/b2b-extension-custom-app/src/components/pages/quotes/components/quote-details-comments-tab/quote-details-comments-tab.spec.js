import React from 'react';
import { render, fireEvent, wait } from '@testing-library/react';
import QuoteDetailsCommentsTab from './quote-details-comments-tab';

jest.mock('react-intl', () => {
  const actual = jest.requireActual('react-intl');
  return {
    ...actual,
    useIntl: () => ({
      formatDate: date => date,
      formatTime: time => time,
      formatMessage: msg => msg.defaultMessage,
    }),
  };
});
jest.mock('@commercetools-frontend/application-shell-connectors', () => {
  const actual = jest.requireActual(
    '@commercetools-frontend/application-shell-connectors'
  );
  return {
    ...actual,
    useApplicationContext: () => ({ user: { email: 'foo@example.com' } }),
  };
});
jest.mock('@commercetools-frontend/actions-global', () => ({
  useShowNotification: () => jest.fn(),
}));
jest.mock('../quote-details-add-comment-form', () => ({ onSubmit }) => (
  <button
    onClick={() => onSubmit({ text: 'foo-text' })}
    data-testid="mock-button"
  />
));

const quote = {
  comments: [
    { id: 'foo', text: 'text', email: 'foo@example.com', createdAt: 'date' },
    { id: 'bar', text: 'text', email: 'bar@example.com', createdAt: 'date' },
  ],
};

describe('QuoteDetailsCommentsTab', () => {
  it('should render the correct snapshot', () => {
    const { asFragment } = render(
      <QuoteDetailsCommentsTab addComment={jest.fn()} quote={quote} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it('should call addComment on submit children', async () => {
    const addCommentSpy = jest.fn();
    const { getByTestId } = render(
      <QuoteDetailsCommentsTab addComment={addCommentSpy} quote={quote} />
    );
    const submitChildren = getByTestId('mock-button');

    fireEvent.click(submitChildren);
    await wait();
    expect(addCommentSpy).toHaveBeenCalledWith({
      email: 'foo@example.com',
      text: 'foo-text',
    });
  });
  it('should call addComent with employeeEmail prop', async () => {
    const addCommentSpy = jest.fn();
    const { getByTestId } = render(
      <QuoteDetailsCommentsTab
        addComment={addCommentSpy}
        quote={quote}
        employeeEmail="internal@example.com"
      />
    );
    const submitChildren = getByTestId('mock-button');

    fireEvent.click(submitChildren);
    await wait();
    expect(addCommentSpy).toHaveBeenCalledWith({
      email: 'internal@example.com',
      text: 'foo-text',
    });
  });
});
