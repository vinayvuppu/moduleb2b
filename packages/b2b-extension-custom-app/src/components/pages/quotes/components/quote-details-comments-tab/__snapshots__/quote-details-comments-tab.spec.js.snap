// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteDetailsCommentsTab should render the correct snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-z1wx52-getStyles"
  >
    <button
      data-testid="mock-button"
    />
    <div
      class="css-ngf9uq-Card"
    >
      <div
        class="css-z1wx52-getStyles"
      >
        <h4
          class="css-78wsr5-subheadlineStyles"
        >
          foo@example.com - date date
        </h4>
        <p
          class="css-f4iurr-bodyStyles"
        >
          <span
            class="pre-wrap"
          >
            text
          </span>
        </p>
      </div>
    </div>
    <div
      class="css-ngf9uq-Card"
    >
      <div
        class="css-z1wx52-getStyles"
      >
        <h4
          class="css-78wsr5-subheadlineStyles"
        >
          bar@example.com - date date
        </h4>
        <p
          class="css-f4iurr-bodyStyles"
        >
          <span
            class="pre-wrap"
          >
            text
          </span>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
